### PR TITLE
Fix JSON validator + correct tides.spec.json for successful validation

### LINKED ISSUE #136: 🐛💻 – table schema validation fails due to old OpenSSL in validator's docker image
## Describe the problem

The `requests` library used by validator isn't pinned and the latest release uses `urllib` version 2, which itself requires OpenSSL >= 1.1.1.  The validator runs on a docker image using python 3.7 on debian buster, both of which are years old and don't include OpenSSL >= 1.1.1 by default.

## Steps to reproduce

The validate tables workflow fails because of this:
https://github.com/TIDES-transit/TIDES/actions/runs/4926209132/jobs/8801438813?pr=132#step:5:19

## Options

- fork the validator and switch the python docker image to python3.11-slim-bullseye
- submit an issue to the validator to update the docker image used, however the project has open issues dating back years and hasn't been updated since August 2020.
- submit an issue to the docker image project https://github.com/docker-library/python to add support for OpenSSL 1.1.1 to the python3.7-slim-buster image used by the current validator.
- switch to a different validator action, not entirely sure what the options are though the Frictionless action only supports data validation, not schema validation.
- develop our own schema validation action or workflow, this requires some work, but doesn't seem very difficult.

### PR DESCRIPTION
**Pull request by SorenSpicknall**
# Pull Request
This PR resolves #136 via a personal fork of the unmaintained GitHub Action we had in use - we can always fork a version directly under the TIDES project later instead of having it live under my GitHub account.

The PR also uses `tides-data-package` to validate `tides.spec.json`, and makes small changes to `tides.spec.json` in order to pass validation.

The merge-ready PR doesn't include a run of the second Action that was changed, defined in `validate_table_schemas.yml`, but that was tested on previous commits via small changes to `operators.schema.json` that have now been undone.

### CODE REVIEW

**Review by e-lo — CHANGES_REQUESTED** [review]
Let's at a min put the image under a non-personal GH account :-)

File: .github/workflows/validate_package_schema.yml [hunk 1]
```diff
@@ -14,10 +14,10 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Validate Data Package schema
-        uses: snapcart/json-schema-validator@v1.0.0
+        uses: SorenSpicknall/json-schema-validator@master
```
[the remark this answers is not in the document]
Comment (SorenSpicknall) [reply .github/workflows/validate_package_schema.yml]: The trouble we have right now is that Frictionless primarily references and uses the JSON schema standard `draft-04`, which is not supported by recent versions of the popular ajv JSON validation tool that underlies most of the available GitHub actions that perform this task. This limits our options for externally supported automations; if we used `draft-07` instead, we could rely on [the well-maintained `json-yaml-validate` Action](https://github.com/marketplace/actions/json-yaml-validate), but that would mean a lot of departure from standard Frictionless stuff.

I will modify this PR to use [the `validate-json` Action](https://github.com/marketplace/actions/validate-json), which relies on an older version of ajv and supports `draft-04`. The downside to that one is we need to manually list each of the JSON files we're validating within the workflow YAML, so if/when somebody adds a new table schema down the road it won't get picked up for validation until the table is referenced in the workflow definition.

**Comment by SorenSpicknall** [comment]
Okay, so here's where we're at:

The most popular actively maintained JSON validator Action doesn't work for our purposes because A.) it uses a version of ajv that doesn't support `draft-04`, and B.) if we upgrade to `draft-07` the validator doesn't give us the option to directly set the ajv setting that controls strict interpretation of unknown keywords - so the validator fails on keywords like `propertyOrder` and `context` due to ajv opinionated contravention of [general JSON Schema implementation guidelines](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-02#section-6.5).

Most of the other validator Actions out there have some version of the same problem. So I forked the original validator into the TIDES GitHub organization, which is flexible enough to work for our purposes but introduces a possible future maintenance burden.

In addition, I went ahead and updated JSON Schema standard referenced by `tides-data-package.json` and `data-schema.json` to use `draft-07` (which doesn't require changes in our schemas, since the only breaking changes from `draft-04` to `draft-07` aren't applicable to our implementation). The intent here is to give us a head start on future alignment with a newer, more widely used version of the standard and enable us to hook into externally maintained validation tools later if one meets our needs. However, since the examples from Frictionless still use `draft-04` and these files pass CI now either way, I can be persuaded to use `draft-04` instead to remain in alignment with those examples for now. Happy to go in either direction based on your input, @e-lo.

**Comment by e-lo** [comment]
@SorenSpicknall wondering if this is actually now moot with new version released just now: https://github.com/frictionlessdata/frictionless-ci/pull/48

**Comment by SorenSpicknall** [comment]
> @SorenSpicknall wondering if this is actually now moot with new version released just now: [frictionlessdata/frictionless-ci#48](https://github.com/frictionlessdata/frictionless-ci/pull/48)

My understanding from the description of #136 and a quick glance at the docs is that the Frictionless-maintained Action is solely geared toward data validation right now, and doesn't support schema validation. That's why we've been using the other Action, `json-schema-validator`. The Dockerfile update that was made to the Frictionless-maintained Action doesn't appear to be directly related to the problems we had with the Action we're using, unless there's something I'm missing.

It's definitely worth continuing to keep the Frictionless tool on the edge of our vision in case they add schema validation in the future, since a community-maintained option that suits our needs would certainly be nicer to work with than the fork used in this PR, but I don't believe the Frictionless-maintained Action supports the functionality we're looking for right now.

**Review by e-lo — APPROVED** [review]
Let's try it out...drumroll...

**Pull request merged**